File: NodeJsProject/tests/msIdentity/dotnetIdentity.test.js
const { validatePassword, registerUser, hashPassword, newUserHashPassword } = require("../../msIdentity/msIdentity")

describe("MS Identity Passwordhash testing", () => {

    test("Check valid password1", () => {

        const hashedPwd = "AQAAAAEAACcQAAAAEHJ76TcZq46JvLgCzCOWf55uupFZesz7ZUg4io6EIrQb5n/2pQM4dPTexgJmeHaZsQ==";
        const password = 'Secret01!'
        const check = validatePassword(password, hashedPwd)

        expect(check).toBe(true);
    });

    test("Check valid password2", () => {

        const hashedPwd = "AQAAAAEAACcQAAAAEFF6kaUfVFtKu8JuTDq5PuuBXLGFIdrLCqUGivmyNiyQfn3WGCWBcr9wtSwJ9A1E7A==";
        const password = 'Test123!'
        const check = validatePassword(password, hashedPwd)

        expect(check).toBe(true);
    });

    test("Check invalid password", () => {

        const hashedPwd = "AQAAAAEAACcQAAAAEFF6kaUfVFtKu8JuTDq5PuuBXLGFIdrLCqUGivmyNiyQfn3WGCWBcr9wtSwJ9A1E7A==";
        const password = 'test123#'
        const check = validatePassword(password, hashedPwd)

        expect(check).toBe(false);
    });


    test("Check invalid password", () => {

        const hashedPwd = "ABAAAAEAACcQAAAAEFF6kaUfVFtKu8JuTDq5PuuBXLGFIdrLCqUGivmyNiyQfn3WGCWBcr9wtSwJ9A1E7A==";
        const password = 'Test123!'
        const check = validatePassword(password, hashedPwd)

        expect(check).toBe(false);
    });

    test("Hash password static hash", () => {
        
        const password = 'NeverGonnaTell!'
        const hashedPwd = hashPassword(password, 'SHA256', 10000, 16, '517A91A51F545B4ABBC26E4C3AB93EEB')

        const check = validatePassword(password, hashedPwd)
        expect(check).toBe(true);
    });

    test("Hash password random hash", () => {
        
        const password = 'NeverGonnaTell!'
        const hashedPwd = newUserHashPassword(password)

        const check = validatePassword(password, hashedPwd)
        expect(check).toBe(true);
    });
})

File: JavaProject/src/test/java/uc/no/identity/DotnetIdentityTest.java
package uc.no.identity;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.Base64;

import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;

import uc.no.identity.msIdentity.MsIdentityService;

public class DotnetIdentityTest {
    
    @Test
    void validatePassword01() {

        String hashedPwd = "AQAAAAEAACcQAAAAEHJ76TcZq46JvLgCzCOWf55uupFZesz7ZUg4io6EIrQb5n/2pQM4dPTexgJmeHaZsQ==";
        String password = "Secret01!";
        boolean res = MsIdentityService.validatePassword(password, hashedPwd);

        assertTrue(res, "Password OK");        
    }

    @Test
    void validatePassword02() {

        String hashedPwd = "AQAAAAEAACcQAAAAEFF6kaUfVFtKu8JuTDq5PuuBXLGFIdrLCqUGivmyNiyQfn3WGCWBcr9wtSwJ9A1E7A==";
        String password = "Test123!";
        boolean res = MsIdentityService.validatePassword(password, hashedPwd);

        assertTrue(res, "Password OK");        
    }  

    @Test
    void checkInvalidPassword01() {

        String hashedPwd = "AQAAAAEAACcQAAAAEFF6kaUfVFtKu8JuTDq5PuuBXLGFIdrLCqUGivmyNiyQfn3WGCWBcr9wtSwJ9A1E7A==";
        String password = "Test123#";
        boolean res = MsIdentityService.validatePassword(password, hashedPwd);

        assertFalse(res, "Password OK");        
    }

    @Test
    void SelfHashedPassword01() {
        String password = "H3mmeligt!";
        String passwordHash = MsIdentityService.newUserPassword(password);

        boolean res = MsIdentityService.validatePassword(password, passwordHash);

        assertTrue(res, "Hashed password checkes out OK");        
    }

    @Test
    void SelfHashedPasswordInvalid01() {
        String password = "H3mmeligt!";
        String passwordHash = MsIdentityService.newUserPassword(password);

        boolean res = MsIdentityService.validatePassword("H3mmeligt#", passwordHash);

        assertFalse(res, "Hashed password checkes out OK");        
    }
}
